plone/app/content/browser/contents/copy.py

```diff
@@ -50,5 +53,38 @@ def finish(self):
         self.request['__cp'] = cp
 
     def __call__(self):
+        self.protect()
         self.oblist = []
-        return super(CopyActionView, self).__call__()
+        self.errors = []
+        context = aq_inner(self.context)
+        selection = self.get_selection()
+
+        parts = str(self.request.form.get('folder', '').lstrip('/')).split('/')
+        if parts:
+            parent = self.site.unrestrictedTraverse('/'.join(parts[:-1]))
+            self.dest = parent.restrictedTraverse(parts[-1])
+
+        self.catalog = getToolByName(context, 'portal_catalog')
+        self.mtool = getToolByName(self.context, 'portal_membership')
+
+        missing = []
+        for uid in selection:
```

Comment:
This is the block which different from the superclass.
Alternatively I could introduce another method for overriding in the superclass and override only the part which iterates over the selected objects and executes actions.
But probably that's introducing unnecessary complexity...
Comments welcome.